### PR TITLE
Added regular roulette

### PR DESCRIPTION
**Pull request by TeeSeal**
So uh.. yeah, regular casino roulette!

I haven't set up an environment to run Commando so I only tested the roulette mechanics, not quite sure how it's going to work with currency and discord yet, which means the code needs good reviewing and some testing.

### CODE REVIEW

File: Roulette.js [hunk 1]
```diff
@@ -34,3 +56,38 @@ module.exports = class Roulette {
 		});
 	}
 };
+
+function generateSpaces() {
```
Comment (iCrawl) [inline Roulette.js]: Any reasons you put those functions outside of the class?

File: commands/games/roulette-info.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+			group: 'games',
+			memberName: 'roulette-info',
+			description: 'Displays information about the roulette.',
+			details: 'Displays information about the roulette.'
+		});
+	}
+
+	async run(msg) {
+		return msg.embed({
+			image: { url: 'http://i.imgur.com/l88QEHm.png' },
```
Comment (iCrawl) [inline commands/games/roulette-info.js]: Should probably use HTTPS and use an imagehost where I know it wont run out, use cuntflaps.me if possible 👌

File: commands/games/roulette.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+const currency = new Currency();
+
+module.exports = class RouletteCommand extends Command {
+	constructor(client) {
+		super(client, {
+			name: 'roulette',
+			aliases: ['roulette', 'bet'],
+			group: 'games',
+			memberName: 'roulette',
+			description: 'Play a game of russian roulette for donuts!',
```
Comment (iCrawl) [inline commands/games/roulette.js]: -russian

File: commands/games/roulette.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+			details: 'Play a game of russian roulette for donuts.',
```
Comment (iCrawl) [inline commands/games/roulette.js]: Ditto.

File: commands/games/roulette.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+			throttling: {
+				duration: 30,
```
Comment (iCrawl) [inline commands/games/roulette.js]: Keep those 30sec? Might wanna set that higher? Not sure, think about how long one round usually takes and take into account how long it needs to cooldown. Cooldown starts at command useage.

File: commands/games/roulette.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+	async run(msg, args) {
+		const bet = args.donuts;
+		const space = args.space.toLowerCase();
+
+		const balance = await currency.getBalance(msg.author.id);
+		let roulette = Roulette.findGame(msg.guild.id);
+
+		if (balance < 100) {
+			return msg.reply(`you don't have enough 🍩s. You need at least 100 🍩s to bet, but your current account balance is ${balance} 🍩s.`);
```
Comment (iCrawl) [inline commands/games/roulette.js]: Do the winnings get split? If they do, please use a number thats easily divideable, and do it like in russian-roulette with 120, 140, 160 or whatever.

File: commands/games/roulette.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+		}
+
+		if (![100, 200, 300, 400, 500].includes(bet)) {
```
Comment (iCrawl) [inline commands/games/roulette.js]: Ditto.

File: commands/games/russian-roulette.js [hunk 8]
```diff
@@ -87,9 +87,4 @@ module.exports = class RouletteCommand extends Command {
 		barrel[Math.floor(Math.random() * barrel.length)] = 1;
 		return barrel;
 	}
-
-	removeDuplicates(array) {
```
Comment (iCrawl) [inline commands/games/russian-roulette.js]: Reason to remove that?

File: commands/games/roulette.js (hunk 6 above)
Comment (TeeSeal) [inline commands/games/roulette.js]: Winnings don't get split. Each user plays individually, with the quirk that multiple users can play at the same time. Said user can bet on only one space in one instance of the game. Upon joining/betting the user loses the donuts they bet. If they win they get their bet * multiplier, if they lose they just lose they get nothing back.

File: commands/games/russian-roulette.js (hunk 8 above)
Comment (TeeSeal) [inline commands/games/russian-roulette.js]: It's used nowhere. Asked Grey about it and he said I could remove it ¯\_(ツ)_/¯

File: Roulette.js (hunk 1 above)
Comment (TeeSeal) [inline Roulette.js]: Thought I didn't need new ones for each instance of the class. Will this not work the way I intended it to?

Comment (iCrawl) [inline Roulette.js]: Oh right. Haven't thought about that.

Comment (iCrawl) [inline Roulette.js]: Oh right, haven't tought about that 👍

File: commands/games/russian-roulette.js (hunk 8 above)
Comment (iCrawl) [inline commands/games/russian-roulette.js]: If it's nowhere used and he said its fine, its 👍

File: commands/games/roulette.js (hunk 6 above)
Comment (iCrawl) [inline commands/games/roulette.js]: Sounds good.

**Comment by iCrawl** [comment]
I'll let Grey review over the rest, since some of the things modify code he wrote 👌 
No other changes requested from my side so far.

**Comment by TeeSeal** [comment]
Sure thing

**Pull request merged**